### DISCUSSION TITLE
Extending LSP capabilities - migrating `kickstart.nvim` config from `nvim-cmp` to `blink.cmp`

### DISCUSSION THREAD

**Discussion by joncrangle** [saghen/blink.cmp#227] [Q&A] [answered]
Sorry for what might be a stupid question.

For those of us starting with `kickstart.nvim` base configs, the LSP capabilities are extended with:
```lua
local capabilities = vim.lsp.protocol.make_client_capabilities()
capabilities = vim.tbl_deep_extend('force', capabilities, require('cmp_nvim_lsp').default_capabilities())
--- ...
      require('mason-lspconfig').setup {
        handlers = {
          function(server_name)
            local server = servers[server_name] or {}
            server.capabilities = vim.tbl_deep_extend('force', {}, capabilities, server.capabilities or {})
            require('lspconfig')[server_name].setup(server)
          end,
        },
      }

```

Is migration from `nvim-cmp` to `blink.cmp` just a matter of changing the 'cmp_nvim_lsp` line to the following?
```lua
capabilities = vim.tbl_deep_extend('force', capabilities, require('blink.cmp').get_lsp_capabilities(capabilities))
```

**Comment by saghen** [comment]
Yeah close, you can do `capabilities = vim.tbl_deep_extend('force', capabilities, require('blink.cmp').get_lsp_capabilities())`. The argument to `get_lsp_capabilities` overrides the properties set by `blink.cmp` so you wouldn't want to override with the defaults.

**Reply by saghen** [reply, marked as answer]
Actually, you can just do `local capabilities = require('blink.cmp').get_lsp_capabilities()` since it'll merge with the default capabilities internally

**Reply by joncrangle** [reply]
Thanks! And thanks a bunch for your work on this plugin.
